components/engine_rocks/src/status.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+pub fn from_rocksdb_error(msg: impl Into<String>) -> engine_traits::Error {
+    // TODO: use correct code.
+    engine_traits::Error::Engine(engine_traits::Status::with_error(
+        engine_traits::Code::IoError,
+        msg,
+    ))
+}
+
+pub fn from_engine_traits_error(s: engine_traits::Error) -> String {
+    format!("{:?}", s)
+}
+
+/// A macro that will transform a rocksdb error to engine trait error.
+///
+/// r stands for rocksdb, e stands for engine_trait.
+macro_rules! r2e {
```

Comment:
can we implement `From` for error and avoid using macro

Reply:
how about adding intermediate type in engine_rocks

Reply:
Then the type needs to be constructed every where. You can't build up such connections that just using `?` to convert two foreign types.

Reply:
how about shortening `from_rocksdb_error` to `r2e`. Writing `xxx().map_err(r2e)` is more readible than `r2e!(xxx())`